runtime/sonic.cpp

```diff
@@ -1101,43 +1101,43 @@ double Sonic_FFT_Filter::filter (
             {
                 // scoot frequencies up...
 
-                for ( i=halfSize; i >= binShift; --i )
+                for ( int i=halfSize; i >= binShift; --i )
                 {
                     freqReal[i] = freqReal[i-binShift];
                     freqImag[i] = freqImag[i-binShift];
                 }
 
-                for ( ; i >= 0; --i )
+                for ( int i=halfSize; i >= 0; --i )

```

Comment:
This initialization is incorrect.  This loop should continue where the previous loop left off.  I think it would be better to have a single 'int i' declaration for the whole function to minimize changes.